polyfill/lib/intl.mjs

```diff
@@ -32,68 +39,45 @@ export function DateTimeFormat(locale = IntlDateTimeFormat().resolvedOptions().l
   this[TIME] = new IntlDateTimeFormat(locale, timeAmend(options));
   this[DATETIME] = new IntlDateTimeFormat(locale, datetimeAmend(options));
 }
+
 DateTimeFormat.supportedLocalesOf = function(...args) {
   return IntlDateTimeFormat.supportedLocalesOf(...args);
 };
+
 const properties = {
-  resolvedOptions: {
-    value: resolvedOptions,
-    enumerable: true,
-    writable: false,
-    configurable: true
-  },
-  format: {
-    value: format,
-    enumerable: true,
-    writable: false,
-    configurable: true
-  },
-  formatRange: {
-    value: formatRange,
-    enumerable: true,
-    writable: false,
-    configurable: true
-  }
+  resolvedOptions: descriptor(resolvedOptions),
+  format: descriptor(format),
+  formatRange: descriptor(formatRange)
 };
+
 if ('formatToParts' in IntlDateTimeFormat.prototype) {
-  properties.formatToParts = {
-    value: formatToParts,
-    enumerable: true,
-    writable: false,
-    configurable: true
-  };
+  properties.formatToParts = descriptor(formatToParts);
 }
+
 if ('formatRangeToParts' in IntlDateTimeFormat.prototype) {
-  properties.formatRangeToParts = {
-    value: formatRangeToParts,
-    enumerable: true,
-    writable: false,
-    configurable: true
-  };
+  properties.formatRangeToParts = descriptor(formatRangeToParts);
 }
+
 DateTimeFormat.prototype = Object.create(IntlDateTimeFormat.prototype, properties);
 
 function resolvedOptions() {
   return this[ORIGINAL].resolvedOptions();
 }
+
 function format(datetime, ...rest) {
   const { absolute, formatter } = extractOverrides(datetime, this);
   if (absolute && formatter) {
     return formatter.format(absolute.getEpochMilliseconds());
   }
-  if (datetime instanceof Duration) {
-    return this[DURATION].format(datetime, ...rest);
-  }
   return this[ORIGINAL].format(datetime, ...rest);
 }
+
 function formatToParts(datetime, ...rest) {
   const { absolute, formatter } = extractOverrides(datetime, this);
   if (absolute && formatter) return formatter.formatToParts(absolute.getEpochMilliseconds());
-  if (datetime instanceof Duration) {
-    return this[DURATION].formatToParts(datetime, ...rest);
-  }
   return this[ORIGINAL].formatToParts(datetime, ...rest);
 }
+
 function formatRange(a, b) {
   if ('object' === typeof a && 'object' === typeof b && Object.getPrototypeOf(a) === Object.getPrototypeOf(b)) {
```

Comment:
```suggestion
  if ('object' === typeof a && 'object' === typeof b && a && b && Object.getPrototypeOf(a) === Object.getPrototypeOf(b)) {
```
otherwise `null` will make it throw.

Reply:
Oops! Fixed. PTAL.